Parser.php

```diff
@@ -82,8 +82,19 @@ class Parser
     public function __construct(PartFactory $partFactory)
     {
         $this->partFactory = $partFactory;
+        $this->reset();
```

Comment:
Yeah, I fucked this class up when I wrote it. There should be a part-class that extends Zends part-class and adds all the part-specific functionality we need (the workarounds for accessing headers, the getPrimaryContent() method, etc).

Also, the parts themselves should be stored in a tree-structure that would make it much easier to identify attachments, forwarded emails and enveloped emails. See the issue I've created for this: https://github.com/lassodatasystems/LassoMailParserBundle/issues/5

However, I don't think it's a good time to re-write the parser. Aside from the potential of introducing new bugs that tests don't cover, I don't think it's a good use of developer time right now.